packages/projection/src/operators/Mappers/withHandles.ts

```diff
@@ -47,17 +83,30 @@ const tryCreateHandleOwnership = (
   if (!policyIds.includes(policyId)) return;
   try {
     const cip67Asset = cip67Assets.byAssetId[assetId];
-    const handle = assetIdToUTF8Handle(assetId, cip67Asset);
+
+    const handle = assetIdToUTF8Handle(assetId);
+    const subhandleProps: Partial<HandleOwnership> = {};
     if (handle) {
+      if (handle.includes('@')) {
+        subhandleProps.parentHandle = handle.split('@')[1];
+
+        if (cip67Asset?.decoded.label === Asset.AssetNameLabelNum.VirtualHandle) {
+          const virtualSubhandleOwnerAddress = getVirtualSubhandleOwnerAddress(txOut?.datum, logger);
+          subhandleProps.latestOwnerAddress = virtualSubhandleOwnerAddress;
+        }
+      }
+
       return {
         assetId,
         datum: txOut?.datum,
         handle,
         latestOwnerAddress: txOut?.address || null,
-        policyId
+        policyId,
+        ...subhandleProps
       };
     }
   } catch (error: unknown) {
+    console.log('ERRROR', error);
```

Comment:
This should be a lint error. No need to duplicate the logs

Reply:
This was a log leftover, it's addressed here: 63a3f045c69a6b818a333afd2a4bb825fe34cf51